metrics/ci_integrations/test/source/buildkite/adapter/buildkite_source_client_adapter_test.dart

```diff
@@ -961,6 +1038,36 @@ void main() {
       },
     );
 
+    test(
+      ".fetchOneBuild() maps in-progress buildkite build to build data with null duration",
```

Comment:
Let's divide this into two test cases: 
- .fetchOneBuild() maps running buildkite build to build data with null duration
- .fetchOneBuild() maps canceling buildkite build to build data with null duration